homeassistant/components/obihai/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow to configure the Obihai integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .connectivity import validate_auth
+from .const import DEFAULT_PASSWORD, DEFAULT_USERNAME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(
+            CONF_USERNAME,
+            default=DEFAULT_USERNAME,
+        ): str,
+        vol.Optional(
+            CONF_PASSWORD,
+            default=DEFAULT_PASSWORD,
+        ): str,
+    }
+)
+
+
+class ObihaiFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Obihai."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            if not await self.hass.async_add_executor_job(
+                validate_auth,
+                user_input[CONF_HOST],
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+            ):
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST],
+                    data=user_input,
+                )
```

Comment:
You can invert the logic here:
```suggestion
            if await self.hass.async_add_executor_job(
                validate_auth,
                user_input[CONF_HOST],
                user_input[CONF_USERNAME],
                user_input[CONF_PASSWORD],
            ):
                return self.async_create_entry(
                    title=user_input[CONF_HOST],
                    data=user_input,
                )
            errors["base"] = "cannot_connect"
```